ibis/sql/compiler.py

```diff
@@ -1023,7 +1023,7 @@ def make_alias(self, expr):
         alias = 't%d' % i
         self.set_ref(expr, alias)
 
-    def need_aliases(self):
+    def need_aliases(self, expr=None):
```

Comment:
Why this change?

Reply:
Because in case of clickhouse it depends on the expression too not just the dialect.

Reply:
Ok. We should document this aspect of the design and see if we can generalize it so that people writing a new backend can ignore this detail if they don't need it.

Reply:
I will do that in my design doc writeup